### PR TITLE
Migrate to TypeScript

### PR DESCRIPTION
**Pull request by LabhanshAgrawal**
Added ts configuration and build toolchain (#2702)
Ported array.js and file.js to TypeScript for demonstration
Incremental adoption can be done one file at a time

### CODE REVIEW

**Comment by Stanzilla** [comment]
That's awesome!

**Comment by chabou** [comment]
Awesome!
I'll review it asap.
Thanks @LabhanshAgrawal for your work and @Stanzilla for your help

File: lib/utils/file.ts [hunk 1]
```diff
@@ -10,7 +10,7 @@
  * PR: https://github.com/kevva/executable/pull/10
  */
 
-export default function isExecutable(fileStat) {
+export default function isExecutable(fileStat: { [k: string]: number }): boolean {
```
Comment (Tyriar) [inline lib/utils/file.ts]: `fileStat: Stats`?

Comment (LabhanshAgrawal) [reply lib/utils/file.ts]: Thanks for pointing it out. Fixed it.

**Comment by LabhanshAgrawal** [comment]
Rebased the branch on latest canary.

**Comment by LabhanshAgrawal** [comment]
Any update on this?

**Comment by Stanzilla** [comment]
I'm not sure how much time @chabou currently has for Hyper but if he hasn't objected in the next few days, I'll merge it

**Comment by Stanzilla** [comment]
Alright, merged! Thank you so much for this!

**Pull request merged**